contract_template/{{cookiecutter.contract_file_name}}/{{cookiecutter.contract_file_name}}.py

```diff
@@ -0,0 +1,96 @@
+# -*- coding: utf-8 -*-
+# ------------------------------------------------------------------------------
+#
+#   Copyright {{cookiecutter.year}} Valory AG
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+# ------------------------------------------------------------------------------
+
+"""This module contains the class to connect to a ERC20 contract."""
```

Comment:
I don't understand why you want to generate all the functions. Why not just have a sufficiently generic base class so it can interpret any ABI? Then all we need is a package with the ABI and a bit of metadata and send messages to it.

Reply:
This current feature is more of an `open-aea` than an `open-autonomy` feature in any case.

Reply:
You mean having just one generic template for any kind of method instead of separating by method type (view, payable...)? Yes that makes a lot of sense actually. And yes, I'll move this over to open-aea.

Reply:
I mean even simpler; at runtime feed the message to the right method (constructed on the fly effectively) and execute the call.

Reply:
Of course, someone might want to add extra methods, so a scaffolded package makes sense with the abi and a very short Contract.py. Then most methods just get auto resolved, the dev can add custom functionality. This approach would also mean there's no need for cookiecutter. Just extend our existing scaffold to handle it.